index.php

```diff
@@ -115,6 +116,13 @@ function failed($reason) {
 require($CFG->dirroot . '/lib/setup.php');
 require_once($CFG->libdir.'/filelib.php');
 
+//IP Locking, check for remote IP in validated list, if not, exit
```

Comment:
this is a bit of a gotcha with this plugin, there are muliple endpoints, ie index.php or croncheck.php etc I would refactor this check into one place and then reuse this across all of them

note that not all the end points need this check, just the ones which expose some sort of info that could be leveraged which I think is just index.php and croncheck.php and maybe upload.php

Reply:
Sure, missed the multiple endpoints